model/src/main/java/org/projectnessie/model/UnreachableHeadsResponse.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.model;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import java.util.List;
+import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.immutables.value.Value;
+
+@Schema(type = SchemaType.OBJECT, title = "UnreachableHeadsResponse")
+@Value.Immutable
+@JsonSerialize(as = ImmutableUnreachableHeadsResponse.class)
+@JsonDeserialize(as = ImmutableUnreachableHeadsResponse.class)
+public interface UnreachableHeadsResponse {
```

Comment:
This should implement paging

Reply:
But the backend always computes a set, it cannot compute iteratively. 

Do you mean, build a stream on top of computed set<Hash> and do paging on that stream?

Reply:
Also, like commit log, backend doesn't persist the computed unreachable heads. So, pagination cannot work based on pageToken?